### PR TITLE
Sharded Plugin 4/n: Expose scaler in amp plugin

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

Related to #4178 

Allows precision plugin to manage grad scaler. This is needed for Sharded as we require a custom scaler object.

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In in short, see following bullet-list:  

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/plugins/native_amp.py [hunk 1]
```diff
@@ -59,3 +59,7 @@ def training_step(self, fx, args):
     def clip_gradients(self, grad_clip_val: Union[int, float], optimizer: Optimizer, norm_type: float):
         model = self.trainer.get_model()
         torch.nn.utils.clip_grad_norm_(model.parameters(), max_norm=grad_clip_val, norm_type=norm_type)
+
+    @property
```
Comment (williamFalcon) [inline pytorch_lightning/plugins/native_amp.py:63]: doesn’t native amp also need this defined? otherwise it will crash when it tries to get the property

Comment (SeanNaren) [reply pytorch_lightning/plugins/native_amp.py:63]: This is native amp, no custom amp yet...

File: pytorch_lightning/trainer/training_loop.py [hunk 2]
```diff
@@ -136,7 +136,7 @@ def setup_training(self, model: LightningModule):
 
         # init amp. Must be done here instead of __init__ to allow ddp to work
         if self.trainer.amp_backend == AMPType.NATIVE and self.trainer.precision == 16 and not self.trainer.use_tpu:
```
Comment (williamFalcon) [inline pytorch_lightning/trainer/training_loop.py:138]: for a future PR let's fold these if statements into accelerators

**Review by williamFalcon — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
Getting there :)

**Review by Borda — COMMENTED** [review]
lgtm :]

**Review by Borda — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]
